tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java

```diff
@@ -2605,6 +2606,219 @@ public static void test_supportsLikeEscapeClause_shouldBe_true() throws Exceptio
 		);
 	}
 	
+	public static void test_supports_catalogs_in_table_definitions() throws Exception {
+		final Connection connection = DriverManager.getConnection("jdbc:duckdb:");
+		final Statement statement = connection.createStatement();
+		final DatabaseMetaData databaseMetaData = connection.getMetaData();
+		final String CATALOG_NAME = "tmp";
+		final String TABLE_NAME = "t1";
+		final String qualifiedTableName = CATALOG_NAME + "." + TABLE_NAME;
+		ResultSet resultSet = null;
+		try {
+			statement.execute("ATTACH '' AS \"" + CATALOG_NAME + "\"");
+
+			final boolean supportsCatalogsInTableDefinitions = databaseMetaData.supportsCatalogsInTableDefinitions();
+			try {
+				statement.execute("CREATE TABLE " + qualifiedTableName + "(id int)");
+			}
+			catch (SQLException ex) {
+				if (supportsCatalogsInTableDefinitions) {
+					fail(
+						"supportsCatalogsInTableDefinitions is true but CREATE TABLE in attached database is not allowed. "+
+						ex.getMessage()
+					);
+					ex.printStackTrace();
+				}
+			}
+			resultSet = statement.executeQuery(
+				"SELECT * FROM information_schema.tables "+
+				"WHERE table_catalog = '" + CATALOG_NAME + "' " +
+				"AND table_name = '" + TABLE_NAME + "' "
+			);
+			assertTrue(resultSet.next(), "Expected exactly 1 row from information_schema.tables, got 0");
+			assertFalse(resultSet.next());
+			resultSet.close();
+
+			try {
+				statement.execute("DROP TABLE " + qualifiedTableName);
+			}
+			catch (SQLException ex) {
+				if (supportsCatalogsInTableDefinitions) {
+					fail(
+						"supportsCatalogsInTableDefinitions is true but DROP TABLE in attached database is not allowed. " + 
+						ex.getMessage()
+					);
+					ex.printStackTrace();
+				}
+			}
+			resultSet = statement.executeQuery(
+				"SELECT * FROM information_schema.tables "+
+				"WHERE table_catalog = '" + CATALOG_NAME + "' " +
+				"AND table_name = '" + TABLE_NAME + "' "
+			);
+			assertTrue(resultSet.next() == false, "Expected exactly 0 rows from information_schema.tables, got > 0");
+			resultSet.close();
+
+			assertTrue(supportsCatalogsInTableDefinitions, "supportsCatalogsInTableDefinitions should return true.");
+		} 
+		finally {
+			if (resultSet != null) {
+				resultSet.close();
+			}
+			statement.close();
```

Comment:
Can you please use try with resources instead of all these explicit close calls?